src/probability/strong_law.lean

```diff
@@ -0,0 +1,752 @@
+/-
+Copyright (c) 2022 Sébastien Gouëzel. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Sébastien Gouëzel
+-/
+
+import probability.ident_distrib
+import measure_theory.function.l2_space
+import measure_theory.integral.interval_integral
+import analysis.specific_limits.floor_pow
+import analysis.p_series
+import analysis.asymptotics.specific_asymptotics
+
+/-!
+# The strong law of large numbers
+
+We prove the strong law of large numbers, in `probability_theory.strong_law_ae`:
+If `X n` is a sequence of independent identically distributed integrable real-valued random
+variables, then `∑ i in range n, X i / n` converges almost surely to `𝔼[X 0]`.
+We give here the strong version, due to Etemadi, that only requires pairwise independence.
+
+## Implementation
+
+We follow the proof by Etemadi, which goes as follows.
+
+It suffices to prove the result for nonnegative `X`, as one can prove the general result by
+splitting a general `X` into its positive part and negative part.
+Consider `Xₙ` a sequence of nonnegative integrable identically distributed pairwise independent
+random variables. Let `Yₙ` be the truncation of `Xₙ` up to `n`. We claim that
+* Almost surely, `Xₙ = Yₙ` for all but finitely many indices. Indeed, `∑ ℙ (Xₙ ≠ Yₙ)` is bounded by
+  `1 + 𝔼[X]` (see `sum_prob_mem_Ioc_le` and `tsum_prob_mem_Ioi_lt_top`).
+* Let `c > 1`. Along the sequence `n = c ^ k`, then `(∑_{i=0}^{n-1} Yᵢ - 𝔼[Yᵢ])/n` converges almost
+  surely to `0`. This follows from a variance control, as
+  ∑_k ℙ (|∑_{i=0}^{c^k - 1} Yᵢ - 𝔼[Yᵢ]| > c^k ε)
+    ≤ ∑_k (c^k ε)^{-2} ∑_{i=0}^{c^k - 1} Var[Yᵢ]    (by Markov inequality)
+    ≤ ∑_i (C/i^2) Var[Yᵢ]                           (as ∑_{c^k > i} 1/(c^k)^2 ≤ C/i^2)
+    ≤ ∑_i (C/i^2) 𝔼[Yᵢ^2]
+    ≤ 2C 𝔼[X^2]                                     (see `sum_variance_truncation_le`)
```

Comment:
I'm afraid that the documentation website will reflow the lines.

Reply:
Yes, I agree. But at least it is readable when browsing the lean file.

Reply:
What if you add ``` before and after?

Reply:
Sure, it can't hurt in any case. It looks better in https://observablehq.com/@bryangingechen/github-lean-doc-preview